GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexEntry.cs

```diff
@@ -105,11 +139,22 @@ public unsafe void ParsePath()
                     int partIndex = this.NumParts;
 
                     byte* forLoopPtr = pathPtr + forLoopStartIndex;
+                    byte* bufferPtr;
+                    int bufferLength;
                     for (int i = forLoopStartIndex; i < this.PathLength + 1; i++)
                     {
                         if (*forLoopPtr == PathSeparatorCode)
                         {
-                            this.PathParts[partIndex] = LazyUTF8String.FromByteArray(pathPtr + currentPartStartIndex, i - currentPartStartIndex);
+                            bufferPtr = pathPtr + currentPartStartIndex;

```

Comment:
This is a very hot code path, and writing these temp values to variables could cause a measurable perf hit if the compiler thinks it has to write it out from a register into memory. Please measure this to make sure we're not hurting perf here. It's possible that moving the variable scope into this inner block will help the compiler figure it out, but in the end you may be better off duplicating the math and not writing to a variable at all.